packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -98,7 +99,9 @@ const AvatarGroup = React.forwardRef(function AvatarGroup(inProps, ref) {
     return React.isValidElement(child);
   });
 
-  const extraAvatars = children.length > clampedMax ? children.length - clampedMax + 1 : 0;
+  const totalAvatars = total || children.length;
+  const avatarsToShow = children.length < clampedMax ? children.length : clampedMax - 1;
```

Comment:
Can't show more avatars than the number available from `children`. I introduced the `avatarsToShow` variable here to make sure that is the case.